controller/config.go

```diff
@@ -600,6 +601,32 @@ func validateSpaceConfig(c Config, key, topic string) error {
 	return nil
 }
 
+// AsSpaceConstraints checks to see whether config has spaces names populated
+// for management and/or HA (Mongo).
+// Non-empty values are merged with any input spaces and returned as a new
+// constraint.Value.
+func (c Config) AsSpaceConstraints(spaces *[]string) constraints.Value {
```

Comment:
Following the foray into replacing the slice pointers with slices in constraints.Value, and the subsequent roll-back, I have used a *[]string in and out here.

I think this makes it work a little cleaner when modifying constraints. Otherwise one would have to get the result check it and set the reference on the target in multiple call sites.